server/planning/planning/planning_history_async_service.py

```diff
@@ -0,0 +1,207 @@
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+"""Superdesk Files"""
+
+import logging
+from copy import deepcopy
+from typing import Any
+
+
+from planning.core.service import PlanningResourceModelType
+from planning.types import PlanningHistoryResourceModel
+from superdesk.flask import request
+from superdesk.resource_fields import ID_FIELD
+from superdesk.default_settings import strtobool
+
+from planning.types import PlanningResourceModel
+from planning.history_async_service import HistoryAsyncService
+from planning.common import WORKFLOW_STATE, ITEM_ACTIONS, ASSIGNMENT_WORKFLOW_STATE
+from planning.item_lock import LOCK_ACTION
+from planning.assignments.assignments_history import ASSIGNMENT_HISTORY_ACTIONS
+from planning.utils import (
+    get_related_event_links_for_planning,
+    is_coverage_planning_modified,
+    is_coverage_assignment_modified,
+)
+
+
+logger = logging.getLogger(__name__)
+update_item_actions = ["assign_agenda", "add_featured", "remove_featured"]
+
+
+class PlanningHistoryAsyncService(HistoryAsyncService[PlanningHistoryResourceModel]):
+    async def on_item_created(self, items: list[dict[str, Any]], operation=None):
+        add_to_planning = False
+        if request and hasattr(request, "args"):
+            add_to_planning = strtobool(request.args.get("add_to_planning", "false"))
+        await super().on_item_created(items, "add_to_planning" if add_to_planning else None)
+
+    async def _save_history(self, item, update: dict[str, Any], operation: str | None = None):
+        user = await self.get_user_id()
+        # confirmation could be from external fulfillment, so set the user to the assignor
+        if operation == ASSIGNMENT_HISTORY_ACTIONS.CONFIRM and user is None:  # type: ignore
```

Comment:
You can convert/create an enum for `ASSIGNMENT_HISTORY_ACTIONS`, use it to type the parameter `operation` and then avoid the `# type: ignore` here.